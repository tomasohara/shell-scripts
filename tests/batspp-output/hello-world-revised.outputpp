#!/usr/bin/env bats


# Make executables ./tests/../ visible to PATH
PATH="/home/aveey/tom-project/shell-scripts/tests/batspp-only/../:$PATH"

# Enable aliases
shopt -s expand_aliases



@test "test-1" {
	testfolder="/tmp/batspp-78052/test-1"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'alias testuser="sed -r "s/"$USER"+/userxf333/g""\n' "=========="
	test-1-actual 
	echo "=========" $'$ alias testnum="sed -r "s/[0-9]/N/g""' "========="
	test-1-expected 
	echo "============================"
	# ???: 'alias testuser="sed -r "s/"$USER"+/userxf333/g""\n'=$(test-1-actual)
	# ???: '$ alias testnum="sed -r "s/[0-9]/N/g""'=$(test-1-expected)
	[ "$(test-1-actual)" == "$(test-1-expected)" ]
}

function test-1-actual () {
	# no-op in case content just a comment
	true

	alias testuser="sed -r "s/"$USER"+/userxf333/g""

}

function test-1-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
$ alias testnum="sed -r "s/[0-9]/N/g""
END_EXPECTED
}


@test "test-3" {
	testfolder="/tmp/batspp-78052/test-3"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'echo "Hi Mom!"\n' "=========="
	test-3-actual 
	echo "=========" $'Hi Mom!' "========="
	test-3-expected 
	echo "============================"
	# ???: 'echo "Hi Mom!"\n'=$(test-3-actual)
	# ???: 'Hi Mom!'=$(test-3-expected)
	[ "$(test-3-actual)" == "$(test-3-expected)" ]
}

function test-3-actual () {
	# no-op in case content just a comment
	true

	echo "Hi Mom!"

}

function test-3-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
Hi Mom!
END_EXPECTED
}


@test "test-4" {
	testfolder="/tmp/batspp-78052/test-4"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $"echo 'Hello world' | wc -l\n" "=========="
	test-4-actual 
	echo "=========" $'1' "========="
	test-4-expected 
	echo "============================"
	# ???: "echo 'Hello world' | wc -l\n"=$(test-4-actual)
	# ???: '1'=$(test-4-expected)
	[ "$(test-4-actual)" == "$(test-4-expected)" ]
}

function test-4-actual () {
	# no-op in case content just a comment
	true

	echo 'Hello world' | wc -l

}

function test-4-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
1
END_EXPECTED
}


@test "test-5" {
	testfolder="/tmp/batspp-78052/test-5"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'pwd | testuser\n' "=========="
	test-5-actual 
	echo "=========" $'/home/userxf333/tom-project/shell-scripts/tests' "========="
	test-5-expected 
	echo "============================"
	# ???: 'pwd | testuser\n'=$(test-5-actual)
	# ???: '/home/userxf333/tom-project/shell-scripts/tests'=$(test-5-expected)
	[ "$(test-5-actual)" == "$(test-5-expected)" ]
}

function test-5-actual () {
	# no-op in case content just a comment
	true

	pwd | testuser

}

function test-5-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
/home/userxf333/tom-project/shell-scripts/tests
END_EXPECTED
}


@test "test-6" {
	testfolder="/tmp/batspp-78052/test-6"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'which systemd\n' "=========="
	test-6-actual 
	echo "=========" $'/usr/bin/systemd' "========="
	test-6-expected 
	echo "============================"
	# ???: 'which systemd\n'=$(test-6-actual)
	# ???: '/usr/bin/systemd'=$(test-6-expected)
	[ "$(test-6-actual)" == "$(test-6-expected)" ]
}

function test-6-actual () {
	# no-op in case content just a comment
	true

	which systemd

}

function test-6-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
/usr/bin/systemd
END_EXPECTED
}


@test "test-7" {
	testfolder="/tmp/batspp-78052/test-7"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'man date | head -n 10 | testnum\n' "=========="
	test-7-actual 
	echo "=========" $'DATE(N)                          User Commands                         DATE(N)' "========="
	test-7-expected 
	echo "============================"
	# ???: 'man date | head -n 10 | testnum\n'=$(test-7-actual)
	# ???: 'DATE(N)                          User Commands                         DATE(N)'=$(test-7-expected)
	[ "$(test-7-actual)" == "$(test-7-expected)" ]
}

function test-7-actual () {
	# no-op in case content just a comment
	true

	man date | head -n 10 | testnum

}

function test-7-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
DATE(N)                          User Commands                         DATE(N)
END_EXPECTED
}


@test "test-8" {
	testfolder="/tmp/batspp-78052/test-8"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'ps -l | awk \'!($3=$4=$5=$10=$13="")\' | testnum\n' "=========="
	test-8-actual 
	echo "=========" $'F S    C PRI NI ADDR  WCHAN TTY  CMD\nN S    N NN N -  do_wai pts/N  bash\nN R    N NN N -  - pts/N  ps\nN S    N NN N -  pipe_r pts/N  awk\nN S    N NN N -  pipe_r pts/N  sed' "========="
	test-8-expected 
	echo "============================"
	# ???: 'ps -l | awk \'!($3=$4=$5=$10=$13="")\' | testnum\n'=$(test-8-actual)
	# ???: 'F S    C PRI NI ADDR  WCHAN TTY  CMD\nN S    N NN N -  do_wai pts/N  bash\nN R    N NN N -  - pts/N  ps\nN S    N NN N -  pipe_r pts/N  awk\nN S    N NN N -  pipe_r pts/N  sed'=$(test-8-expected)
	[ "$(test-8-actual)" == "$(test-8-expected)" ]
}

function test-8-actual () {
	# no-op in case content just a comment
	true

	ps -l | awk '!($3=$4=$5=$10=$13="")' | testnum

}

function test-8-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
F S    C PRI NI ADDR  WCHAN TTY  CMD
N S    N NN N -  do_wai pts/N  bash
N R    N NN N -  - pts/N  ps
N S    N NN N -  pipe_r pts/N  awk
N S    N NN N -  pipe_r pts/N  sed
END_EXPECTED
}


@test "test-9" {
	testfolder="/tmp/batspp-78052/test-9"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'ps | head -n 10 | testuser | testnum\n' "=========="
	test-9-actual 
	echo "=========" $'PID TTY          TIME CMD\n  NNNNN pts/N    NN:NN:NN bash\n  NNNNN pts/N    NN:NN:NN ps\n  NNNNN pts/N    NN:NN:NN head\n  NNNNN pts/N    NN:NN:NN sed\n  NNNNN pts/N    NN:NN:NN sed' "========="
	test-9-expected 
	echo "============================"
	# ???: 'ps | head -n 10 | testuser | testnum\n'=$(test-9-actual)
	# ???: 'PID TTY          TIME CMD\n  NNNNN pts/N    NN:NN:NN bash\n  NNNNN pts/N    NN:NN:NN ps\n  NNNNN pts/N    NN:NN:NN head\n  NNNNN pts/N    NN:NN:NN sed\n  NNNNN pts/N    NN:NN:NN sed'=$(test-9-expected)
	[ "$(test-9-actual)" == "$(test-9-expected)" ]
}

function test-9-actual () {
	# no-op in case content just a comment
	true

	ps | head -n 10 | testuser | testnum

}

function test-9-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
PID TTY          TIME CMD
  NNNNN pts/N    NN:NN:NN bash
  NNNNN pts/N    NN:NN:NN ps
  NNNNN pts/N    NN:NN:NN head
  NNNNN pts/N    NN:NN:NN sed
  NNNNN pts/N    NN:NN:NN sed
END_EXPECTED
}
