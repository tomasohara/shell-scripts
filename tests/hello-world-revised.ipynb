{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cc34e95a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD\n",
    "## shopt -s expand_aliases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "220201d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/userxf333/g\"\"\n",
    "alias testnum=\"sed -r \"s/[0-9]/N/g\"\"\n",
    "\n",
    "## For testnum being non-critical\n",
    "# alias testnum=\"sed -r \"s/[0-9][0-9]*/X/g\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "10e4aac0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hi Mom!\n"
     ]
    }
   ],
   "source": [
    "echo \"Hi Mom!\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "93dfcf88",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "echo 'Hello world' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "95585f5b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "userxf333\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "# pwd | testuser\n",
    "\n",
    "whoami | testuser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1c7c7b9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# THIS IS A COMMENT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7e5dcc7c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/bin/systemd\n"
     ]
    }
   ],
   "source": [
    "# SKIP\n",
    "which systemd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a3e68607",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "# man date | head -n 10\n",
    "## Focus on pattern matching\n",
    "date | grep --perl-regexp --count --ignore-case '^[A-Z]{3}.*[0-9]{4}$'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ec969a63",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "F S    C PRI NI ADDR  WCHAN TTY  CMD\n",
      "N S    N NN N -  do_wai pts/N  bash\n",
      "N R    N NN N -  - pts/N  ps\n",
      "N S    N NN N -  pipe_r pts/N  awk\n",
      "N S    N NN N -  pipe_r pts/N  sed\n"
     ]
    }
   ],
   "source": [
    "# awk will be used to remove any factors that may cause an error\n",
    "ps -l | awk '!($3=$4=$5=$10=$13=\"\")' | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "95a43fbc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    PID TTY          TIME CMD\n",
      "  NNNNN pts/N    NN:NN:NN bash\n",
      "  NNNNN pts/N    NN:NN:NN ps\n",
      "  NNNNN pts/N    NN:NN:NN head\n",
      "  NNNNN pts/N    NN:NN:NN sed\n",
      "  NNNNN pts/N    NN:NN:NN sed\n"
     ]
    }
   ],
   "source": [
    "ps | head -n 10 | testuser | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "038ec2ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "## Zero or Non-Zero checking for wc -l reduces non-critical tests\n",
    "ps | { [ $(wc -l < /dev/stdin) -ne 0 ] && echo $?; }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4b05bb35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".py files present\n"
     ]
    }
   ],
   "source": [
    "## grep-echo pipeline filter: Return true if \"character\" present else false\n",
    "## Example: Check if there are python files (py) in testfolders\n",
    "ls | grep -q \"py\" && echo \".py files present\" || echo \"No .py files present\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ef05beae",
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Incase of a false condition, echo $? returns nothing\n",
    "ls | grep -q \"pyx\" && echo $?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
