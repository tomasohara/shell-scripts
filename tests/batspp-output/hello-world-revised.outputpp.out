1..8
not ok 1 test-1
# (in test file batspp-output/hello-world-revised.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== alias testuser="sed -r "s/"$USER"+/userxf333/g""
#  ==========
# ========= $ alias testnum="sed -r "s/[0-9]/N/g"" =========
# $ alias testnum="sed -r "s/[0-9]/N/g""
# ============================
ok 2 test-3
ok 3 test-4
not ok 4 test-5
# (from function `test-5-actual' in file batspp-output/hello-world-revised.outputpp, line 134,
#  in test file batspp-output/hello-world-revised.outputpp, line 121)
#   `test-5-actual ' failed with status 127
# ========== pwd | testuser
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/hello-world-revised.outputpp: line 134: testuser: command not found
ok 5 test-6
not ok 6 test-7
# (from function `test-7-actual' in file batspp-output/hello-world-revised.outputpp, line 202,
#  in test file batspp-output/hello-world-revised.outputpp, line 189)
#   `test-7-actual ' failed with status 127
# ========== man date | head -n 10 | testnum
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/hello-world-revised.outputpp: line 202: testnum: command not found
not ok 7 test-8
# (from function `test-8-actual' in file batspp-output/hello-world-revised.outputpp, line 236,
#  in test file batspp-output/hello-world-revised.outputpp, line 223)
#   `test-8-actual ' failed with status 127
# ========== ps -l | awk '!($3=$4=$5=$10=$13="")' | testnum
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/hello-world-revised.outputpp: line 236: testnum: command not found
not ok 8 test-9
# (from function `test-9-actual' in file batspp-output/hello-world-revised.outputpp, line 274,
#  in test file batspp-output/hello-world-revised.outputpp, line 261)
#   `test-9-actual ' failed with status 127
# ========== ps | head -n 10 | testuser | testnum
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/hello-world-revised.outputpp: line 274: testuser: command not found
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/hello-world-revised.outputpp: line 274: testnum: command not found
